src/Microsoft.SqlTools.ServiceLayer/LanguageServices/PeekDefinition.cs

```diff
@@ -0,0 +1,150 @@
+        public PeekDefinition(ConnectionInfo connInfo)
+        {
+            connectionInfo = connInfo;
+            tempPath = Path.GetTempPath();
+        }
+
+        /// <summary>
+        /// Script a table using SMO and write to a file.
+        /// </summary>
+        /// <param name="tableName">Table name</param>
+        /// <returns>Location object representing URI and range of the script file</returns>
+        internal Location[] GetTableDefinition(string tableName)
+        {
+            if (this.connectionInfo.SqlConnection != null)
+            {
+                Server server = new Server(this.connectionInfo.SqlConnection.DataSource);
+                Database database = server.Databases[this.connectionInfo.SqlConnection.Database];
+                Table table = database.Tables[tableName];
+                string tempFileName = tempPath + tableName + ".sql";
+
+                if (table != null)
+                {
+                    using (StreamWriter scriptFile = new StreamWriter(File.Open(tempFileName, FileMode.Create, FileAccess.ReadWrite)))
+                    {
+                        StringCollection scripts = table.Script();
+                        foreach (string script in scripts)
+                        {
+                            if (script.Contains("CREATE TABLE"))
+                            {
+                                scriptFile.WriteLine(script);
+                            }  
+                        }
+
+                        
+                    }
+                    return GetLocationFromFile(tempFileName);
+                }
+            }
+            return null;
+        }
+
+        /// <summary>
+        /// Script a view using SMO and write to a file.
+        /// </summary>
+        /// <param name="viewName">View name</param>
+        /// <param name="schemaName">Schema name </param>
+        /// <returns>Location object representing URI and range of the script file</returns>
+        internal Location[] GetViewDefinition(string viewName, string schemaName)
+        {
+            if (this.connectionInfo.SqlConnection != null)
+            {
+                Server server = new Server(this.connectionInfo.SqlConnection.DataSource);
+                Database database = server.Databases[this.connectionInfo.SqlConnection.Database];
+                View view = (schemaName != null) ? database.Views[viewName, schemaName] : database.Views[viewName];
+                string tempFileName = (schemaName != null) ? tempPath + schemaName + "." + viewName + ".sql"
+                                                    : tempPath + viewName + ".sql";
+
+                if (view != null)
+                {
+                    using (StreamWriter scriptFile = new StreamWriter(File.Open(tempFileName, FileMode.Create, FileAccess.ReadWrite)))
+                    {
+                        StringCollection scripts = view.Script();
+                        foreach (string script in scripts)
```

Comment:
Same comment as above regarding simplifying using LINQ and what happens if there are more than one "CREATE VIEW"